### ISSUE TITLE
Sidecar container lifecycle changes since Kubernetes 1.18

### ISSUE THREAD

**Issue by char-iot** [helm/helm#12193]
Output of `helm version`:
version.BuildInfo{Version:"v3.12.1", GitCommit:"f32a527a060157990e2aa86bf45010dfb3cc8b8d", GitTreeState:"clean", GoVersion:"go1.20.4"}

Output of `kubectl version  --output=yaml`:
clientVersion:
  buildDate: "2023-05-17T14:20:07Z"
  compiler: gc
  gitCommit: 7f6f68fdabc4df88cfea2dcf9a19b2b830f1e647
  gitTreeState: clean
  gitVersion: v1.27.2
  goVersion: go1.20.4
  major: "1"
  minor: "27"
  platform: linux/amd64
kustomizeVersion: v5.0.1
serverVersion:
  buildDate: "2023-05-18T15:22:01Z"
  compiler: gc
  gitCommit: 890a139214b4de1f01543d15003b5bda71aae9c7
  gitTreeState: clean
  gitVersion: v1.26.5+rke2r1
  goVersion: go1.19.9 X:boringcrypto
  major: "1"
  minor: "26"
  platform: linux/amd64

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
RKE2

basically when installing a chart with a job which has a sidecar container lifecycle job won't finish.
```bash
helm install --namespace helm-test -f values.yaml helm-test1 ./testapp
```
```yaml
apiVersion: batch/v1
kind: Job
metadata:
  name: {{ include "testapp.fullname" . }}
  annotations:
    helm.sh/hook: "pre-install, pre-upgrade"
spec:
  template:
    spec:
      containers:
        - image: nginx:latest
          lifecycle:
            type: Sidecar
```
```
Release "helm-test1" does not exist. Installing it now.
W0711 13:09:27.923456    4500 warnings.go:70] unknown field "spec.template.spec.containers[1].lifecycle.type"
```

nice to have, but also it's from 1.18, it'd be nice for helm to have it too

**Comment by wujunwei** [comment]
It seems that there is no field named `type` under `Lifecycle` , could you show me some document about it ?

**Comment by philomory** [comment]
This appears to be based on a mistaken belief that a proposed-but-not-actually-accepted feature back from 1.18 got implemented; even the name of the issue is very similar to the title of an article on the subject: [Sidecar container lifecycle changes in Kubernetes 1.18](https://banzaicloud.com/blog/k8s-sidecars/)
